lib/Connection.js

```diff
@@ -1421,6 +1421,13 @@ Connection.prototype._resUntagged = function(info) {
         msgEmitter = new EventEmitter();
         attrs = {};
 
+        let infoTextKeys = (info.text && Object.keys(info.text)) || [];
+        // ignore unsolicted untagged flag-update fetch responses
+        if (!(infoTextKeys.length <= 3 && infoTextKeys.length > 0 &&
+            !infoTextKeys.some(k => ['flags','uid','modseq'].indexOf(k) === -1))) {
+          this._curReq.bodyEmitter.emit('message', msgEmitter, info.num);
+        }
+
```

Comment:
This is going to emit the same data again even if the `if` was hit?

Reply:
The condition is if it's not the unsolicited message. I'll update the comment to say:
if this unsolicited, untagged  fetch response is not a flag-update, emit a message event